pkgs/by-name/po/postlight-parser/package.nix

```
@@ -1,36 +1,53 @@
 { lib
 , stdenv
-, mkYarnPackage
 , fetchFromGitHub
 , fetchYarnDeps
+, yarnConfigHook
+, yarnBuildHook
+, nodejs
+, npmHooks
 }:
 
-mkYarnPackage rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "postlight-parser";
   version = "2.2.3";
 
   src = fetchFromGitHub {
     owner = "postlight";
     repo = "parser";
-    rev = "v${version}";
+    rev = "v${finalAttrs.version}";
     hash = "sha256-k6m95FHeJ+iiWSeY++1zds/bo1RtNXbnv2spaY/M+L0=";
   };
 
-  packageJSON = ./package.json;
-
-  doDist = false;
-
   offlineCache = fetchYarnDeps {
-    yarnLock = "${src}/yarn.lock";
+    yarnLock = "${finalAttrs.src}/yarn.lock";
     hash = "sha256-Vs8bfkhEbPv33ew//HBeDnpQcyWveByHi1gUsdl2CNI=";
   };
 
-  meta = with lib; {
-    changelog = "https://github.com/postlight/parser/blob/${src.rev}/CHANGELOG.md";
+  nativeBuildInputs = [
+    yarnConfigHook
+    yarnBuildHook
+    nodejs
+    npmHooks.npmInstallHook
+  ];
+  # Upstream doesn't include a script in package.json that only builds without
+  # testing, and tests fail because they need to access online websites. Hence
+  # we use the builtin interface of yarnBuildHook to lint, and in `postBuild`
+  # we run the rest of commands needed to create the js files eventually
+  # distributed and wrapped by npmHooks.npmInstallHook
+  yarnBuildScript = "lint";
+  postBuild = ''
+    yarn --offline run rollup -c
+  '';
+  # Tries to download stuff from the internet in this phase.
+  dontNpmPrune = true;
+
```

```suggestion
  dontStrip = true;

```
I'm not sure if this is something that will apply to other packages, but at least here:
```
stripping (with command strip and flags -S) in  /nix/store/84pydmm10lh5061grgv3c4509wkffzgk-postlight-parser-2.2.3/lib /nix/store/84pydmm10lh5061grgv3c4509wkffzgk-postlight-parser-2.2.3/bin
fixupPhase completed in 1 minutes 58 seconds
```
The stripping takes ages looking at all the JS files which don't need it.

---

> The stripping takes ages looking at all the JS files which don't need it.

I agree, but this is a general comment about the default `fixupPhase` of `node` related projects. Is there a particular reason you focused here on `postlight-master`? Perhaps in general we should set `stripExclude` in node projects... In anycase, I think this discussion is out of scope for this PR. The issue you mentioned was present with `mkYarnPackage` as well.

---

> Is there a particular reason you focused here on postlight-master?

I'm the maintainer for this package, so this is the one I tested :)